modules/linear_boltzmann_solvers/a_lbs_solver/source_functions/source_function.cc

```diff
@@ -17,19 +17,19 @@ void
 SourceFunction::operator()(LBSGroupset& groupset,
                            std::vector<double>& destination_q,
                            const std::vector<double>& phi_local,
-                           SourceFlags source_flags)
+                           SourceFlags source)
 {
-  if (source_flags & NO_FLAGS_SET) return;
+  if (source.Empty()) return;
 
   const size_t source_event_tag = lbs_solver_.GetSourceEventTag();
   Chi::log.LogEvent(source_event_tag, chi::ChiLog::EventType::EVENT_BEGIN);
 
-  apply_fixed_src_ = (source_flags & APPLY_FIXED_SOURCES);
-  apply_wgs_scatter_src_ = (source_flags & APPLY_WGS_SCATTER_SOURCES);
-  apply_ags_scatter_src_ = (source_flags & APPLY_AGS_SCATTER_SOURCES);
-  apply_wgs_fission_src_ = (source_flags & APPLY_WGS_FISSION_SOURCES);
-  apply_ags_fission_src_ = (source_flags & APPLY_AGS_FISSION_SOURCES);
-  suppress_wg_scatter_src_ = (source_flags & SUPPRESS_WG_SCATTER);
+  apply_fixed_src_ = (source & APPLY_FIXED_SOURCES);
+  apply_wgs_scatter_src_ = (source & APPLY_WGS_SCATTER_SOURCES);
+  apply_ags_scatter_src_ = (source & APPLY_AGS_SCATTER_SOURCES);
+  apply_wgs_fission_src_ = (source & APPLY_WGS_FISSION_SOURCES);
+  apply_ags_fission_src_ = (source & APPLY_AGS_FISSION_SOURCES);
+  suppress_wg_scatter_src_ = (source & SUPPRESS_WG_SCATTER);
```

Comment:
```suggestion
  apply_fixed_src_ = (source_flags & APPLY_FIXED_SOURCES);
  apply_wgs_scatter_src_ = (source_flags & APPLY_WGS_SCATTER_SOURCES);
  apply_ags_scatter_src_ = (source_flags & APPLY_AGS_SCATTER_SOURCES);
  apply_wgs_fission_src_ = (source_flags & APPLY_WGS_FISSION_SOURCES);
  apply_ags_fission_src_ = (source_flags & APPLY_AGS_FISSION_SOURCES);
  suppress_wg_scatter_src_ = (source_flags & SUPPRESS_WG_SCATTER);
```

Reply:
I think it is clearer to actually keep `source_flags` here. (same remark for all other comments below). I am ready to commit the suggestion (and the ones below) but I'll wait until tomorrow to see if someone disagrees with Zach's proposed changes.